meerkat_libs/db_adapters/postgresql.py

```diff
@@ -0,0 +1,246 @@
+import psycopg2
+import logging
+import json
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+from psycopg2 import sql
+from psycopg2.extras import Json
+
+
+class PostgreSQLAdapter():
+
+    def __init__(self, db_name, structure, user='postgres', host='db'):
+        self.db_name = db_name
+        self.structure = structure
+        self.user = user
+        self.host = host
+        self.keys = self._extract_keys()
+
+    def drop(self):
+        try:
+            logging.info("Dropping tables")
+            conn = psycopg2.connect(
+                dbname=self.db_name,
+                user=self.user,
+                host=self.host
+            )
+
+            for table in self.structure.keys():
+                conn.cursor().execute("DROP TABLE {};".format(table))
+                conn.commit()
+
+            conn.close()
+
+        except psycopg2.ProgrammingError:
+            logging.info("No tables exists to drop.")
+
+    def setup(self):
+        self._create_db()
+        self._create_tables()
+
+    def list(self):
+        for table in self.structure.keys():
+            logging.info("Listing all records in {} table".format(table))
+            cur = self.conn().cursor()
+            cur.execute(sql.SQL("SELECT data from {}").format(
+                sql.Identifier(table)
+            ))
+            results = [x[0] for x in cur.fetchall()]
+            cur.close()
+            logging.info(results)
+
+    def _extract_keys(self):
+        keys = {}
+        for table, structure in self.structure.items():
+            keys[table] = [x[0] for x in structure if x[0] != 'data']
+        return keys
+
+    def _create_db(self):
+        try:
+            logging.info("Checking if db exist.")
+            self.conn = psycopg2.connect(
+                dbname=self.db_name,
+                user=self.user,
+                host=self.host
+            )
+            logging.info("DB exists.")
+
+        except psycopg2.OperationalError:
+            logging.info("DB needs to be created.")
```

Comment:
Here's the actual db creation. I think it should be a separate method called `_create_db` and this one could be named `create_db_if_needed`.